--- a/src/multimedia/windows/qwindows_propertystore.cpp	2026-05-03 11:57:48.285087734 +0200
+++ b/src/multimedia/windows/qwindows_propertystore.cpp	2026-05-03 11:58:28.725994364 +0200
@@ -72,31 +72,6 @@
     return std::nullopt;
 }
 
-namespace {
-
-// mingw workaround: PropVariantToGUID does not provide correct signature
-// https://github.com/mingw-w64/mingw-w64/issues/153
-
-template <typename T>
-struct first_arg_is_ptr;
-
-// Function type
-template <typename Result, typename Arg, typename... Args>
-struct first_arg_is_ptr<Result (*)(Arg, Args...)> : std::bool_constant<std::is_pointer_v<Arg>>
-{
-};
-
-auto wrapPropVariantArg(const PROPVARIANT &arg)
-{
-    if constexpr (first_arg_is_ptr<decltype(&PropVariantToGUID)>::value) {
-        return &arg;
-    } else {
-        return arg;
-    }
-}
-
-} // namespace
-
 std::optional<QUuid> PropertyStoreHelper::getGUID(const PROPERTYKEY &property)
 {
     PROPVARIANT variant;
@@ -106,7 +81,7 @@
         return std::nullopt;
 
     GUID ret;
-    HRESULT hr = PropVariantToGUID(wrapPropVariantArg(variant), &ret);
+    HRESULT hr = PropVariantToGUID(variant, &ret);
     if (SUCCEEDED(hr))
         return QUuid{ ret };
 
